Automattic/newspack-plugin
# Bring PWA feature management in-house

### All Submissions:






### Changes proposed in this Pull Request:



This PR removes Progressive WP as the plugin we use for managing PWA features and brings that functionality into the Newspack plugin (see PWA class).

As part of this work I've simplified the wizard down to one screen, as the only thing that needs to be managed is a site icon:

<img width="654" alt="Screen Shot 2019-08-31 at 8 17 39 AM" src="https://user-images.githubusercontent.com/7317227/64065978-77c86e80-cbc9-11e9-9656-903f9ffa2cdc.png">


### How to test the changes in this Pull Request:

1. You'll need https for this to work properly. I tested with jurassic.ninja, but if you have https on your local environment that would work also.
2. Navigate to the Performance screen in the Newspack dashboard. You should observe a screen like the one pictured above, with a useful status message and ability to upload a site icon.
3. In chrome://flags, [enable these settings](https://user-images.githubusercontent.com/7317227/63796123-46465f00-c8ba-11e9-8d32-7a5b33439313.png).
4. On the frontend, you should see a (+) and have ability to install the site:
<img width="398" alt="Screen Shot 2019-08-31 at 8 18 11 AM" src="https://user-images.githubusercontent.com/7317227/64066033-2371be80-cbca-11e9-9e84-aea21a0a284e.png">

5. In dev tools, go to Applications and toggle offline on:
<img width="375" alt="Screen Shot 2019-08-31 at 8 17 08 AM" src="https://user-images.githubusercontent.com/7317227/64066035-28367280-cbca-11e9-8bfa-9743bd7c6fca.png">

6. Visit a page you've already visited. It should be available. Visit a page you haven't visited; it should give PWA's default offline message for now (until our theme has an `offline.php` template).

### Other information:
